examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -224,33 +247,44 @@ export class TaskList extends DataObject implements ITaskList {
 
         // TODO: Delete any items that are in the root but missing from the external data
         const updateTaskPs = updatedExternalData.map(async ([id, { name, priority }]) => {
-            const currentTask = this.draftData.get<PersistedTask>(id);
             // Write external data into savedData map.
-            this.savedData.set(id, currentTask);
-
+            const currentTask = this.savedData.get<PersistedTask>(id);
+            // Create a new task because it doesn't exist already
             if (currentTask === undefined) {
+                const savedNameString = SharedString.create(this.runtime);
+                const savedPriorityCell = SharedCell.create(this.runtime) as ISharedCell<number>;
+                const savedDataPT: PersistedTask = {
+                    id,
+                    name: savedNameString.handle as IFluidHandle<SharedString>,
+                    priority: savedPriorityCell.handle as IFluidHandle<ISharedCell<number>>,
+                };
+                savedNameString.insertText(0, name);
+                savedPriorityCell.set(priority);
+                this.savedData.set(id, savedDataPT);
+            } else {
+                // Make changes to exisiting saved tasks
+                const [savedNameString, savedPriorityCell] = await Promise.all([
+                    currentTask.name.get(),
+                    currentTask.priority.get(),
+                ]);
+                savedNameString.insertText(0, name);
+                savedPriorityCell.set(priority);
+            }
+
+            // Now look for differences between draftData and savedData
+            const task = this.tasks.get(id);
+            if (task === undefined) {
                 // A new task was added from external source, add it to the Fluid data.
                 this.addTask(id, name, priority);
                 return;
             }
-            const [currName, currPriority] = await Promise.all([
-                currentTask.name.get(),
-                currentTask.priority.get(),
-            ]);
-            if (currName.getText() !== name) {
-                // TODO: Currently replacing existing Fluid data.  But eventually this is where
-                // we'd want conflict resolution UX.
-                currName.replaceText(0, currName.getLength(), name);
+            if (task.name.getText() !== name) {
```

Comment:
Instead of comparing the name here, perhaps it would be better to check above (while writing the new saved data in) whether the new saved data is different from the old saved data above (a change was made remotely) and then additionally whether the current Fluid name is different from the old saved data (a change was made locally).  We only have to negotiate the diff if both of these have happened, otherwise we should be able to resolve silently.

Reply:
If both these cases haven't happened, do we still want to flag that the data has come in overwritten what was there visually to the client? Or should it be silent in the sense that they don't even know that it has happened?